pkg/controller/queuejob/queuejob_controller_ex.go

```diff
@@ -2269,8 +2270,19 @@ func (cc *XController) manageQueueJob(qj *arbv1.AppWrapper, podPhaseChanges bool
 				klog.V(10).Infof("[Dispatcher Controller] Dispatched AppWrapper %s to Agent ID: %s.", qj.Name, agentId)
 				if cc.serverOption.ExternalDispatch {
 					values := strings.Split(agentId, "/")
-					klog.V(10).Infof("[Dispatcher Controller] Dispatching AppWrapper %s to Agent ID: %s Through External Dispatcher.", qj.Name, values[len(values)-1])
-					qj.Status.TargetClusterName = values[len(values)-1] //agentId
+					klog.V(10).Infof("[Dispatcher Controller] ... Dispatching AppWrapper %s to Agent ID: %s Through External Dispatcher.", qj.Name, values[len(values)-1])					
+					qj.Spec.SchedSpec.ClusterScheduling.PolicyResult = arbv1.ClusterDecision {
+						TargetCluster: arbv1.ClusterReference{
+							Name: values[len(values)-1],
```

Comment:
I think there is a better way of obtaining the target cluster id that what is done here by the previous development.  Can you add a `TODO` for merging into opensource main branch to check for values > 0 and assigning the id from the first index of the values?  Just the TODO is fine for a merge of this code.  Thanks!